code/modules/makeup/nailpolish.dm

```diff
@@ -0,0 +1,158 @@
+/obj/item/organ/external/var/datum/nail_polish/nail_polish
+
+/obj/item/weapon/nailpolish
+	name = "nail polish"
+	desc = "Nail polish, to paint your nails with. Or someone else's!"
+	icon = 'icons/obj/nailpolish_vr.dmi'
+	icon_state = "nailpolish"
+	var/colour = "#FF0000"
+	var/icon/top_underlay
+	var/icon/top_underlay_open
+	var/icon/color_underlay
+	var/icon/color_underlay_open
+	var/open = FALSE
+	drop_sound = 'sound/items/drop/glass.ogg'
+	pickup_sound = 'sound/items/pickup/glass.ogg'
+
+/obj/item/weapon/nailpolish/Initialize()
+	. = ..()
+	regenerate_icons()
+
+/obj/item/weapon/nailpolish/proc/regenerate_icons()
+	top_underlay = icon(icon, "top")
+	top_underlay_open = icon(icon, "top-open")
+	color_underlay = icon(icon, "color")
+	color_underlay.Blend(colour, ICON_MULTIPLY)
+	color_underlay_open = icon(icon, "color-open")
+	color_underlay_open.Blend(colour, ICON_MULTIPLY)
+	update_icon()
+
+/obj/item/weapon/nailpolish/proc/set_colour(var/_colour)
+	colour = _colour
+	regenerate_icons()
+
+/obj/item/weapon/nailpolish/attack_self(var/mob/user)
+	open = !open
+	to_chat(user, SPAN_NOTICE("You [open ? "open" : "close"] \the [src]."))
+	update_icon()
+
+/obj/item/weapon/nailpolish/update_icon()
+	. = ..()
+	underlays.Cut()
+	icon_state = "[initial(icon_state)][open ? "-open" : null]"
+	if (open)
+		underlays = list(color_underlay_open, top_underlay_open)
+	else
+		underlays = list(color_underlay, top_underlay)
+
```

Comment:
Why underlays and not overlays? Would be nice if SSoverlays was used for this so the time the appearance churn takes can be accounted for by the MC.

Doing `add_overlay(list(thing1,thing2))` is also atomic since SSoverlays queues it and handles it all at once.